Source/Core/Inline.cs

```diff
@@ -242,31 +290,36 @@ protected void ResolveImpl(Implementation impl)
     // override this and implement their own inlining policy
     protected virtual int GetInlineCount(CallCmd callCmd, Implementation impl)
     {
-      return GetInlineCount(impl);
+      return TryDefineCount(callCmd, impl);
     }
 
-    // returns true if it is ok to further unroll the procedure
-    // otherwise, the procedure is not inlined at the call site
-    protected int GetInlineCount(Implementation impl)
+    protected int TryDefineCount(CallCmd callCmd, Implementation impl)
     {
       Contract.Requires(impl != null);
       Contract.Requires(impl.Proc != null);
 
-      string /*!*/
-        procName = impl.Name;
-      Contract.Assert(procName != null);
-      if (recursiveProcUnrollMap.TryGetValue(procName, out var c))
+      // getDepth returns -1 when depth for this impl is not defined
+      var depth = depthTracker.getDepth(impl);
+      if (depth >= 0)
       {
-        return c;
+        return depth;
       }
 
-      c = -1; // TryGetValue above always overwrites c
-      impl.CheckIntAttribute("inline", ref c);
-      // procedure attribute overrides implementation
-      impl.Proc.CheckIntAttribute("inline", ref c);
+      int countCall (CallCmd cmd) {
```

Comment:
This method name doesn't seem very clear to me. Maybe `callInlineDepth` or something like that would be clearer?